src/Indigo/Contracts/HomebaseLite/Impl/Vote.hs

```diff
@@ -57,25 +57,26 @@ verifyMinBalance :: HomebaseLiteEntrypoint [BalanceResponseItem]
 verifyMinBalance bris = do
   description [itu|
     Callback for the `balance_of` CPS view of FA2,
-    used to check the balance of a `participant` that used the `propose` entrypoint.
-    Takes as input the list of tuples containing the `owner` , `token_id` and `balance` .
+    used to check the governance token balance of a `participant` that used the
+    `propose` entrypoint.
+    Takes the list of tuples containing the `owner` , `token_id` and `balance` as input.
     |]
   forEach bris \bri -> do
     when (bri #! #briBalance < storage #! #sConfiguration #! #cMinimumBalance) do
       failCustomNoArg @() #notEnoughTokens
 
 [errorDoc| "notEnoughTokens" exception "Some of the `balance`s in the list are
 less than `minimum_balance`"|]
-[errorDoc| "duplicateProposal" exception "Fail if a proposal for the same IPFS
-URI was already submitted before"|]
-[errorDoc| "emptyChoices" bad-argument "The number of choices `n` is `0`."|]
-[errorDoc| "noFA2Contract" contract-internal "Configured FA2 contract not found."|]
+[errorDoc| "duplicateProposal" exception "A proposal for the same IPFS URI already exists"|]
+[errorDoc| "emptyChoices" bad-argument "The proposal has no choices, i.e. the provided list of
+available choices is empty"|]
+[errorDoc| "noFA2Contract" contract-internal "Configured FA2 contract not found"|]
 
 vote :: HomebaseLiteEntrypoint ("proposal_uri" :! URI, "choice_index" :! Natural)
 vote par = do
   description [itu|
     Called by a `participant` to cast a vote on a proposal.
-    Takes as input the proposal's IPFS URI and the choice made `x` .
+    Takes a proposal's IPFS URI and the index of a choice (zero-based).
```

Comment:
```suggestion
    Takes the proposal's IPFS URI and the index of the choice (zero-indexed).
```

Reply:
You could argue for the first "the", but definitely not for the second, since it's _a_ choice, not _the_ choice, i.e. there are potentially multiple possible choices, and we don't know which one from the context.

"Zero-indexed" for an index sounds questionable. Is the index indexed?

Reply:
I've tried to rephrase a bit in 93c06ab78d7b49e819f809dd27466aef383b2d39. I still maintain that "the choice" sounds weird without additional context.

Reply:
> not for the second, since it's a choice, not the choice

:thinking: well it may not be _the choice_, but it's _the choice made_ (as it was phrased before, which I figured using "the" would point at).

I mean, likewise it does take _a proposal URI_ technically, but in a call in particular it takes _the chosen proposal's URI_, no?

What do you think of:
```
Takes as input the proposal's IPFS URI and the index of the selected choice (zero-indexed).
```
instead?

Reply:
The latter is only a suggestion tho, your current
```
Takes the proposal's IPFS URI and a choice index (zero-based).
```
sounds correct to me, so I'm not gonna insist :slightly_smiling_face: 

Since I have no other comment left I'll approve already and let you decide :ok_hand:

Reply:
Let's just leave it as it is then, I regrettably lack the focus to come up with anything better at the moment.